go/tasks/pluginmachinery/k8s/plugin.go

```diff
@@ -91,3 +91,8 @@ type Plugin interface {
 	// Properties desired by the plugin
 	GetProperties() PluginProperties
 }
+
+// Defines functions to clean up the resource object after the task has been completed or cancelled.
+type PluginCleanupPolicy interface {
+	OnAbort(ctx context.Context, kubeClient client.Client, resource client.Object) error
```

Comment:
I would rather keep this signature consistent with Finalize/Abort... we will likely want to send additional things besides kubeClient (potentially the task name/project/domain...) so it's a good pattern to send a `Context` object instead...

Reply:
so just have the signature be `OnAbort(ctx context.Context, tCtx pluginsCore.TaskExecutionContext)` ?

Reply:
I was thinking something like this... (with docs and perhaps better names :-D)
```
type PluginAbortOverride interface {
    PreAbort(ctx context.Context, tCtx pluginsCore.TaskExecutionContext, resource client.Object) (behavior AbortBehavior, err error)
}

type AbortBehavior struct {
    Resource client.Object
    DeleteResource    bool
    Update                  UpdateObjectOperation
    Patch                     PatchObjectOperation
}

type PatchObjectOperation struct {
    Patch     client.Patch
    Options []client.PatchOption
}

type UpdateObjectOperation struct {
    Options []client.UpdateOption
}
```